wolfcrypt/src/random.c

```diff
@@ -1459,7 +1459,8 @@ int wc_Entropy_Get(int bits, unsigned char* entropy, word32 len)
 
         if (ret == 0) {
             /* Condition noise value down to 32-bytes or less. */
-            ret = Entropy_Condition(entropy, entropy_len, noise, noise_len);
+            ret = Entropy_Condition(entropy, entropy_len, (byte*) noise,
```

Comment:
Truncate during the conditioning of the data as opposed to during the raw data gathering